src/Simplify.cpp

```diff
@@ -587,6 +602,19 @@ class Simplify : public IRMutator {
             // Broadcast + Broadcast
             expr = Broadcast::make(mutate(broadcast_a->value - broadcast_b->value),
                                  broadcast_a->width);
+        } else if (cast_a && cast_b &&
+                   cast_a->type == cast_b->type &&
```

Comment:
cast_a->type == cast_b->type is not necessary - you can't construct a Sub node with mismatched types